### PR TITLE
Added colored mosaic cat test to tgo.  Fixes #5356.

### PR DESCRIPTION
**Pull request by AgoinsUSGS**
Truthdata for this cat test is located in: /work/users/agoins/TGO_TRUTHDATA/coloredMos

### CODE REVIEW

File: isis/src/tgo/tsts/coloredMosaic/Makefile [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	$(SPICE) $(TSTARGS) from=$(OUTPUT)/$\$$\1.cub \
+           SPKPREDICTED=true CKPREDICTED=true -batchlist=$(OUTPUT)/bluBasenames.lis > /dev/null;
+	$(LS) $(OUTPUT)/*-BLU-*.cub > $(OUTPUT)/bluCubes.lis;
+
+	# Ingest Pan
+	$(LS) $(INPUT)/*-PAN-*.xml > $(OUTPUT)/panInputs.lis;
+	$(SED) 's/^.*input\///g' $(OUTPUT)/panInputs.lis > $(OUTPUT)/panRelative.lis;
+	$(SED) 's/\.xml//g' $(OUTPUT)/panRelative.lis > $(OUTPUT)/panBasenames.lis;
+	$(INGEST) $(TSTARGS) from=$(INPUT)/\$$\1.xml \
+					 to=$(OUTPUT)/$\$$\1.cub -batchlist=$(OUTPUT)/panBasenames.lis > /dev/null;
```
Comment (jessemapel) [inline isis/src/tgo/tsts/coloredMosaic/Makefile]: Spacing is all over the place across this makefile. It's probably your editor doing things you don't expect. Really egregious things like this line should be cleaned up.

File: isis/src/tgo/tsts/coloredMosaic/Makefile [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	$(LS) $(OUTPUT)/*panEqui.cub > $(OUTPUT)/panMosaic.lis;
+	$(MOSAIC) $(TSTSARGS) \
+   fromlist=$(OUTPUT)/panMosaic.lis \
+   mosaic=$(OUTPUT)/panCassisMosaic.cub \
+   grange='USER' \
+   MAXLAT = `$(GETKEY) from=$(OUTPUT)/equi.map keyword=MaximumLatitude grpname=Mapping`  \
+   MINLAT = `$(GETKEY) from=$(OUTPUT)/equi.map keyword=MinimumLatitude grpname=Mapping`  \
+   MAXLON = `$(GETKEY) from=$(OUTPUT)/equi.map keyword=MaximumLongitude grpname=Mapping` \
+   MINLON = `$(GETKEY) from=$(OUTPUT)/equi.map keyword=MinimumLongitude grpname=Mapping` > /dev/null;
+	$(EXPORT) $(TSTSARGS) from=$(OUTPUT)/panCassisMosaic.cub to=$(OUTPUT)/panCassisMosaic  \
```
Comment (jessemapel) [inline isis/src/tgo/tsts/coloredMosaic/Makefile]: This test shouldn't worry about exporting single color mosaics. It should only export the final color mosaic

File: isis/src/tgo/tsts/coloredMosaic/Makefile [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    MINLON = `$(GETKEY) from=$(OUTPUT)/equi.map keyword=MinimumLongitude grpname=Mapping` > /dev/null;
+		$(EXPORT) $(TSTSARGS) from=$(OUTPUT)/nirCassisMosaic.cub to=$(OUTPUT)/nirCassisMosaic \
+              pdsversion=pds4 > /dev/null;
+
+# Create full color mos
+	$(LS) $(OUTPUT)/*CassisMosaic.cub > $(OUTPUT)/mosaicList.txt;
+	$(CUBEIT) fromlist=$(OUTPUT)/mosaicList.txt to=$(OUTPUT)/coloredMosaic.cub > /dev/null;
+	$(EXPORT) $(TSTSARGS) from=$(OUTPUT)/coloredMosaic.cub to=$(OUTPUT)/coloredMosaic \
+             pdsversion=pds4 > /dev/null;
+
```
Comment (jessemapel) [inline isis/src/tgo/tsts/coloredMosaic/Makefile]: Make sure you clean up the output `coloredMosaic.xml` file.

**Review by jessemapel — CHANGES_REQUESTED** [review]

**Comment by AgoinsUSGS** [comment]
Changes have been implemented

File: isis/src/tgo/tsts/coloredMosaic/Makefile [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	$(RM) $(OUTPUT)/panCassisMosaic.cub;
+	$(RM) $(OUTPUT)/redCassisMosaic.cub;
+
+
+	$(RM) $(OUTPUT)/*CassisMosaic.xml;
+	$(RM) $(OUTPUT)/*CassisMosaic1.txt;
+	$(RM) $(OUTPUT)/*CassisMosaic2.txt;
+	$(RM) $(OUTPUT)/*CassisMosaic3.txt;
+	$(RM) $(OUTPUT)/coloredMosaic.img;
+	$(RM) $(OUTPUT)/coloredMosaic.xml;
```
Comment (jessemapel) [inline isis/src/tgo/tsts/coloredMosaic/Makefile]: Don't remove the output IMG and XML files.  The XML file will need to be modified to remove some things and renamed to a TXT file.

**Comment by jessemapel** [comment]
Couple more changes. See the changes on my branch:

https://github.com/AgoinsUSGS/ISIS3/compare/coloredmos...jessemapel:colormosaic

Also, I'm getting slightly different mosaic extents when I run it locally.  See `/work/users/jmapel/ISIS3/isis/src/tgo/tsts/coloredMosaic` and compare the output to the truth data.

**Review by jessemapel — DISMISSED** [review]

**Review by jessemapel — APPROVED** [review]

**Pull request merged**